noxfile.py

```diff
@@ -173,13 +173,24 @@ def lint(session: nox.sessions.Session):
         A `nox.sessions.Session` object.
 
     """
+    import yaml
+
     # Pip install the session requirements.
     session.install("pre-commit")
-    # Execute the pre-commit linting tasks.
-    cmd = ["pre-commit", "run", "--all-files"]
-    hooks = ["black", "blacken-docs", "flake8", "isort"]
-    for hook in hooks:
-        session.run(*cmd, hook)
+
+    # Load the pre-commit configuration YAML file.
+    with open(".pre-commit-config.yaml", "r") as fi:
+        config = yaml.load(fi, Loader=yaml.FullLoader)
+
+    # Enumerate singleton pre-commit hooks.
```

Comment:
Please can you stick a comment on explaining _why_ we're limiting ourselves to singletons? Thanks

Reply:
Done :+1: 

That said... (sorry for stringing this out :stuck_out_tongue_winking_eye:) looking at the `pre-commit/pre-commit-hooks`, they all actually seem like reasonable hooks to run except for `no-commit-to-branch`.

We could simply opt to have a hook id exclude list, just containing `no-commit-to-branch`, and run everything but that hook. It'll get us away for the weak-ish approach of only including `id`s from a `repo` with **one** `hook`.

Whay say ye?

Reply:
I'm up for that, particularly given you've said that we have some devs who rely on the Nox session for their checks. Not sure if the correct session name would still be `lint` in that case! I'll hold off merging.